Lib/fontTools/designspaceLib/__init__.py

```diff
@@ -1322,3 +1322,14 @@ def loadSourceFonts(self, opener, **kwargs):
                 loaded[source.path] = source.font
             fonts.append(source.font)
         return fonts
+
+    def closeSourceFonts(self):
+        for source in self.sources:
+            if source.font is not None:
+                source.font.close()
```

Comment:
defcon Font objects don't have a close method... SourceDescriptor.font can be UFO objects too. You may want to check if hasattr(source.font, "close")...